packages/salesforce-adapter/test/filters/add_parent_to_record_triggered_flows.test.ts

```diff
@@ -0,0 +1,120 @@
+ */
+
+import { CORE_ANNOTATIONS, Element, ReferenceExpression } from '@salto-io/adapter-api'
+import { buildElementsSourceFromElements } from '@salto-io/adapter-utils'
+import { mockTypes } from '../mock_elements'
+import { createCustomObjectType, defaultFilterContext } from '../utils'
+import { createInstanceElement } from '../../src/transformers/transformer'
+import { OPPORTUNITY_METADATA_TYPE } from '../../src/constants'
+import { FilterWith } from './mocks'
+import filterCreator from '../../src/filters/add_parent_to_record_triggered_flows'
+import { buildFetchProfile } from '../../src/fetch_profile/fetch_profile'
+
+const mockLogError = jest.fn()
+jest.mock('@salto-io/logging', () => ({
+  ...jest.requireActual<{}>('@salto-io/logging'),
+  logger: jest.fn().mockReturnValue({
+    warn: jest.fn((...args) => mockLogError(...args)),
+    debug: jest.fn((...args) => mockLogError(...args)),
+  }),
+}))
+
+describe('addParentToRecordTriggeredFlows', () => {
+  describe('onFetch', () => {
+    let elements: Element[]
+    let elementsSource: Element[]
+    let updateOpportunityFlow: Element
+    let updateLeadFlow: Element
+    let opportunity: Element
+    let lead: Element
+
+    describe('when there are record triggered flows', () => {
+      describe('when addParentToRecordTriggeredFlows in Enabled', () => {
+        describe('when all objects that trigger flows exist in the workspace', () => {
+          beforeEach(async () => {
+            opportunity = createCustomObjectType(OPPORTUNITY_METADATA_TYPE, {})
+            elementsSource = [opportunity]
+            elements = [
+              (updateOpportunityFlow = createInstanceElement(
+                { fullName: 'UpdateOpportunity', start: { object: OPPORTUNITY_METADATA_TYPE } },
+                mockTypes.Flow,
+              )),
+              (updateLeadFlow = createInstanceElement(
+                { fullName: 'UpdateLead', start: { object: 'Lead' } },
+                mockTypes.Flow,
+              )),
+              (lead = createCustomObjectType('Lead', {})),
+            ]
+            const filter: FilterWith<'onFetch'> = filterCreator({
+              config: {
+                ...defaultFilterContext,
+                fetchProfile: buildFetchProfile({
+                  fetchParams: { target: [], optionalFeatures: { addParentToRecordTriggeredFlows: true } },
+                }),
+                elementsSource: buildElementsSourceFromElements(elementsSource),
+              },
+            }) as FilterWith<'onFetch'>
+            await filter.onFetch(elements)
+          })
+          it('should add parent annotation to updateOpportunityFlow', async () => {
+            expect(updateOpportunityFlow.annotations[CORE_ANNOTATIONS.PARENT][0]).toEqual(
+              new ReferenceExpression(opportunity.elemID, opportunity),
+            )
+          })
+          it('should add parent annotation to updateLeadFlow', async () => {
```

Comment:
```suggestion
          it('should add parent annotation when the parent was not fetched in the current partial fetch', async () => {
            expect(updateOpportunityFlow.annotations[CORE_ANNOTATIONS.PARENT][0]).toEqual(
              new ReferenceExpression(opportunity.elemID, opportunity),
            )
          })
          it('should add parent annotation when the parent was fetched in the current partial fetch ', async () => {
```
Tests should describe the tested scenario and not the tested instance names.